react/AvailabilitySubscriber.tsx

```diff
@@ -0,0 +1,209 @@
+ * when a product becomes available again
+ */
+function AvailabilitySubscriber(props: Props) {
+  const productContext = useProduct()
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [emailError, setEmailError] = useState(false)
+  const [didBlurEmail, setDidBlurEmail] = useState(false)
+
+  const [addToAvailabilitySubscriber, { loading, error, data }] = useMutation(
+    ADD_TO_AVAILABILITY_SUBSCRIBER_MUTATION
+  )
+
+  const intl = useIntl()
+
+  const commertialOffer =
+    productContext.selectedItem?.sellers[0]?.commertialOffer
+
+  const available = props.available ?? isAvailable(commertialOffer)
+  const skuId = props.skuId ?? productContext.selectedItem?.itemId
+
+  // Render component only if product is out of sales
+  if (available || !skuId) {
+    return null
+  }
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const variables = {
+      acronym: 'AS',
+      document: {
+        fields: [
+          {
+            key: 'skuId',
+            value: skuId,
+          },
+          {
+            key: 'name',
+            value: name,
+          },
+          {
+            key: 'email',
+            value: email,
+          },
+          {
+            key: 'notificationSend',
+            value: 'false',
+          },
+          {
+            key: 'createdAt',
+            value: new Date().toISOString(),
+          },
+          {
+            key: 'sendAt',
+            value: null,
+          },
+        ],
+      },
+    }
+
+    const result = await addToAvailabilitySubscriber({
+      variables,
+    })
+
+    if (!result.errors) {
+      setName('')
+      setEmail('')
+    }
+
+    const event = new CustomEvent('message:success', {
+      detail: {
+        success: true,
+        message: intl.formatMessage({
+          id: 'store/availability-subscriber.added-message',
+        }),
+      },
+    })
+
+    document.dispatchEvent(event)
+  }
+
+  const validateEmail = (newEmail: string) => {
+    const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+    setEmailError(!emailRegex.test(newEmail.toLowerCase()))
+  }
+
+  const handleEmailChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value)
+    validateEmail(e.target.value)
+  }
+
+  const handleNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value)
+  }
+
+  let emailErrorMessage = ''
+
+  if (didBlurEmail && emailError) {
+    emailErrorMessage = intl.formatMessage({
+      id: 'store/availability-subscriber.invalid-email',
+    })
+  }
+
+  const isFormDisabled = name === '' || email === '' || emailError || loading
+
+  return (
+    <div className={styles.subscriberContainer}>
+      <div className={`${styles.title} t-body mb3`}>
+        {intl.formatMessage({ id: 'store/availability-subscriber.title' })}
```

Comment:
Use `<FormattedMessage />` in every message that doesn't use variables. This will avoid those warnings coming from the `messages` builder when we link this app.

Reply:
No need for this, it will get formatMessage function calls too!

Reply:
Also: https://formatjs.io/docs/guides/advanced-usage#imperative-apis